C/c4Document.cc

```diff
@@ -57,9 +58,14 @@ struct C4DocumentInternal : public C4Document {
 
     void init() {
         docID = _versionedDoc.docID();
-        flags = (C4DocumentFlags)_versionedDoc.flags();
-        if (_versionedDoc.exists())
-            flags = (C4DocumentFlags)(flags | kExists);
+        if(c4doc_isExpired(_db, docID)) {
```

Comment:
The regular flags should be irrelevant if the kExpired flag is set really, and I'm not currently aware of any situation in which you would want a document without knowing that it is expired.  That being said, just because I can't think of one doesn't mean it doesn't exist and this may work just as well as two separate functions.